packages/eas-cli/src/project/projectUtils.ts

```diff
@@ -140,19 +152,20 @@ export async function getProjectIdAsync(
     return localProjectId;
   }
 
-  // Set the project ID if it is missing.
   const projectDir = await findProjectRootAsync(findProjectRootOptions);
   if (!projectDir) {
     throw new Error('Run this command inside a project directory.');
   }
-  const newExp = await setProjectIdAsync(projectDir, options);
 
-  const newLocalProjectId = newExp?.extra?.eas?.projectId;
-  if (!newLocalProjectId) {
-    // throw if we still can't locate the projectId
-    throw new Error('Could not retrieve project ID from app.json');
+  const projectId = await fetchProjectIdFromServerAsync(exp);
+
+  try {
+    await saveProjectIdToAppConfigAsync(projectDir, projectId, options);
+  } catch (e: any) {
+    Log.warn(`Failed to save EAS project ID to app config: ${e.message}`);
```

Comment:
We should probably say something more about that so people clearly know they have to take an action on that. Also things like what is it used for, is the project ID static/secret, etc. Maybe make a FYI guide here https://github.com/expo/fyi and link to that in the warning?

Reply:
It would also be helpful if this requirement is mentioned in https://docs.expo.dev/versions/latest/config/app/#dynamic-configuration-with-appconfigjs). Maybe an extra column for required or optional properties.

Reply:
Brent once posted some good guidance on error messages. I forget what he wrote exactly but it was something like: say what's wrong, why it happened, and what to do about it next.

Reply:
Added a comment to the code, but a large set of instructions will have just been printed out by the time this hits so I just changed this to just print whatever is in the warning.